docs/src/main/asciidoc/telemetry-opentracing-to-otel-tutorial.adoc

```diff
@@ -0,0 +1,504 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+[id="telemetry-opentracing-to-otel-tutorial"]
+= Migrate from OpenTracing to OpenTelemetry tracing
+:categories: observability
+:diataxis-type: tutorial
+include::_attributes.adoc[]
+:topics: observability,opentracing,opentelemetry,tracing,migration
+:extensions: io.quarkus:quarkus-smallrye-opentracing,io.quarkus:quarkus-opentelemetry
+
+Migrate an application from xref:opentracing.adoc.adoc[OpenTracing] to xref:opentelemetry.adoc[OpenTelemetry tracing] in Quarkus 3.x.
+
+The legacy OpenTracing framework has been deprecated in favor of the new OpenTelemetry tracing framework. We announced the https://quarkus.io/blog/quarkus-observability-roadmap-2023/#opentracing-archived[OpenTracing deprecation on November 2022], and we are dropping the extension from core Quarkus and move it to the Quarkiverse.
+
+It is now time to migrate your application to OpenTelemetry tracing if you haven’t done it yet.
+
+If you need to migrate to Quarkus 2.16.x please beware that configuration properties are different and you should check the older Quarkus OpenTelemetry guide version, https://quarkus.io/version/2.16/guides/opentelemetry#configuration-reference[here].
+
+== Prerequisites
+
+include::{includes}/prerequisites.adoc[]
+
+== Summary
+
+The demo has 5 parts. Please read the summary and then jump to the section that best fits your use case.
+
+1 - The *starting point* presents the quickstart app that uses OpenTracing
+
+2 - The first part is good for anyone performing a *big bang change* of OpenTracing when you don't have any manual instrumentation
+
+3 - This is the *big bang replacement* of OpenTracing when you have manually instrumented the code. We explain the main differences between OpenTracing and OpenTelemetry
+
+4 - The last part uses the *OpenTracing shim*. This is useful if you have a large application with manually instrumented the code. It can help performing the migration step by step because it allows the use of the legacy OpenTracing API on top of new OpenTelemetry API
+
+5 - Conclusion and additional resources
+
+The tasks described bellow fall into 3 categories:
+
+* Dependencies
+* Configuration
+* Code
+
+[[starting-point]]
+== Starting point
+
+The base project was created with Quarkus code generator. At some point that generator will not be able to create projects by using the legacy `quarkus-smallrye-opentracing` extension anymore. For that reason, you can checkout this demo project from GitHub:
+
+
+=== Checkout the existing legacy project
+
+[source,bash]
+----
+git clone git@github.com:brunobat/opentracing-quickstart.git
```

Comment:
Where should I place it?

Reply:
Can you initiate a move to the quarkusio org? I think it would be better this way. I let you do that and adjust the URL accordingly and we should be good.

FYI, I moved the Jaeger and SmallRye OpenTracing extensions to the Quarkiverse. I will wait for 3.6.0 core to be released to release them so that they only appear for 3.6+.

I will proceed with the PR to remove them as soon as this is in as I will need to adjust your guide in this PR (the opentracing.adoc doc link).

Reply:
Done